server/users/rfc5802/doc.go

```diff
@@ -0,0 +1,19 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rfc5802
+
+// This package is meant to implement the functions defined in RFC 5802. As such, function names will be non-standard
+// compared to most other names in the project.
+// https://datatracker.ietf.org/doc/html/rfc5802
```

Comment:
It would be nice to understand what this package's relationship to `github.com/xdg-go` is. Naively I would assume that package has a full implementation we could use directly wherever necessary. Right now there's only a single use (stringprep), and I don't have a clear sense if that's just the part you're implementing right now or if there's some other reason you're not using that library for more things. If you're going to wrap it for whatever reason, please summarize what's being wrapped and the motivations for doing so. Could also be in method docs where appropriate.

The other mysterious piece is how `golang.org/x/crypto/pbkdf2` fits into the puzzle. Basically: I would like a high level overview of what this package is adding on top of two libraries being wrapped. And if it's more than just the RFC implementation (is it?) then another package name is probably called for.

Finally, this is in `server/users/rfc5802`. Seems like we probably want `servers/auth` somewhere, probably independently of the `users` package. Things to think about as you flesh this out.

Reply:
This is just implementing RFC 5802 to serve the needs of Postgres. The libraries that are being called are all in service of that. Things like `pbkdf2` are mentioned in the RFC, so I'm using it in the function it's mentioned in. I'm not too sure how else I'd document this XD